core/modules/system/js/modules.js

```diff
@@ -90,10 +90,15 @@ Backdrop.behaviors.moduleFilter = {
       }
     }
 
-    // Clear out the input field when clicking the reset button.
+    // Clear out the input field and search query when clicking the reset
+    // button.
     function resetModuleList(e) {
+      // Clear the input field.
       $input.val('').triggerHandler('keyup');
       e.preventDefault();
+
+      // Clear the search query.
+      window.location.href = window.location.origin + window.location.pathname;
```

Comment:
I _think_ this will cause the page to reload? Setting `window.location` directly will only not reload the page if modifying the anchor. Modifying the query string will cause a reload.

We can use [History.pushState()](https://developer.mozilla.org/en-US/docs/Web/API/History/pushState#change_a_query_parameter) to remove the query string instead. A bit more verbose but I think it does the job without reloading:

```
var currentUrl = new URL(window.location);
currentUrl.searchParams.delete('search');
window.history.replaceState({}, '', currentUrl);
```

Reply:
This is similar to my original approach but looks simpler and cleaner. Done